docs/UserGuide.md

```diff
@@ -41,146 +38,415 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+  Edits the description of the 2nd task to be `buy groceries`.
+
+### 2.12. Deleting a task : `delete`
+
+Deletes the specified task from the task manager.
+Format: `delete INDEX`
+
+> Deletes the task at the specified `INDEX`.
+> The index refers to the index number shown in the most recent listing.
+> The index **must be a positive integer** 1, 2, 3, ...
 
 Examples:
 
-* `edit 1 p/91234567 e/johndoe@yahoo.com`<br>
-  Edits the phone number and email address of the 1st person to be `91234567` and `johndoe@yahoo.com` respectively.
+* `list`
+  `delete 2`
+  Deletes the 2nd task in the list.
+* `find Essay`
+  `delete 1`
+  Deletes the 1st task in the results of the `find` command.
+
+### 2.13. Undo : `undo`
+
+Reverses the previous action done by the user
+
+Format: `undo`
+
+> * Only reverses operations that manipulate data, i.e. `add`, `edit` and `delete`
 
-* `edit 2 Betsy Crower t/`<br>
-  Edits the name of the 2nd person to be `Betsy Crower` and clears all existing tags.
 
-### 2.5. Finding all persons containing any keyword in their name: `find`
+### 2.14. Redo : `redo`
+
+Reverses the previous undo operation
+
+Format: `redo`
+
+> * Only reverses the previous `undo` operation
+
+### 2.15. Finding all tasks containing the keyword in any of their properties: `find`
+
+Finds tasks whose properties contain any of the given keywords.
 
-Finds persons whose names contain any of the given keywords.<br>
 Format: `find KEYWORD [MORE_KEYWORDS]`
 
-> * The search is case sensitive. e.g `hans` will not match `Hans`
-> * The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
-> * Only the name is searched.
-> * Only full words will be matched e.g. `Han` will not match `Hans`
-> * Persons matching at least one keyword will be returned (i.e. `OR` search).
-    e.g. `Hans` will match `Hans Bo`
+> * The search is **NOT** case sensitive. e.g `write` will match `Write`
+> * The order of the keywords does not matter. e.g. `Write Essay` will match `Essay Write`
+> * All properties are searched e.g. `DUE_DATE`, `TAGS`, `DESCRIPTION` etc.
+> * Only full words will be matched e.g. `Ess` will not match `Essay`
+> * Tasks matching at least one keyword will be returned (i.e. `OR` search).
+    e.g. `Essay` will match `Essay Writing Competition`
 
 Examples:
 
-* `find John`<br>
-  Returns `John Doe` but not `john`
-* `find Betsy Tim John`<br>
-  Returns Any person having names `Betsy`, `Tim`, or `John`
+* `find Essay`
+  Returns `Write Essay` and `essay writing`
+* `find Essay Writing Competition`
+  Returns any task having keywords `Essay`, `Writing`, or `Competition`
+* `find 17/7/17`
+  Returns any task having the date `17/7/17`
 
-### 2.6. Deleting a person : `delete`
+### 2.16. Find undone tasks : `find undone`
 
-Deletes the specified person from the address book. Irreversible.<br>
-Format: `delete INDEX`
+Finds undone tasks whose properties contain any of the given keywords.
+
+Format: `find undone KEYWORD [MORE_KEYWORDS]`
```

Comment:
Yes, you can certainly find within a particular scope.  Consider finding free slots within a time period also. This is very useful.